packages/next/src/lib/metadata/is-metadata-route.ts

```diff
@@ -61,9 +105,10 @@ export function isMetadataRouteFile(
   // When strictlyMatchExtensions, the dynamic extension is skipped but
   // static extension is kept, which is usually used for matching route path.
   const trailingMatcher = (strictlyMatchExtensions ? '' : '?') + '$'
-  // Match the optional variants like /opengraph-image2, /icon-a102f4.png, etc.
-  const variantsMatcher = '\\d?'
-  // The -\w{6} is the suffix that normalized from group routes;
+  // Match the optional variants like /icon1.png, /icon2.png, etc.
+  const variantsMatcher = '\\d*'
```

Comment:
`*` will match infinite numbers, we don't support that many posibilities, only 0-9 or without number

Reply:
Yes, you don't need that many icons in real world and it will cause a lot work for entries discovery and HMR